tcl/org.testeditor.tcl.dsl.ide.tests/src/org/testeditor/tcl/dsl/ide/TclContentProposalProviderTest.xtend

```diff
@@ -0,0 +1,370 @@
+			
+			* some step
+			Component: GreetingApplication
+			- Type |
+		'''.withPackage
+
+		// when
+		val proposals = tclSnippet.proposals
+
+		// then
+		proposals => [
+			expect('boolean "true" into <element>').prefix.assertEquals('')
+			expect('"value" into <element> and wait "1"').prefix.assertEquals('')
+			expect('"value" into <element>').prefix.assertEquals('')
+			expect('confidential "param" into <element>').prefix.assertEquals('')
+			reject('Stop application') // proposal must NOT include templates with different prefix
+		]
+	}
+
+	@Test
+	def void testComponentIncompleteMultipartTemplateProposal() {
+		// given
+		var tclSnippet = '''
+			# Some
+			
+			* some step
+			Component: GreetingApplication
+			- Start app|
+		'''.withPackage
+
+		// when
+		val proposals = tclSnippet.proposals
+
+		// then
+		proposals => [
+			expect('application "path"').prefix.assertEquals('app')
+			reject('Stop application') // proposal must NOT include templates with different prefix
+		]
+	}
+
+	@Test
+	def void testComponentDenormalizedIncompleteMultipartTemplateProposal() {
+		// given
+		var tclSnippet = '''
+			# Some
+			
+			* some step
+			Component: GreetingApplication
+			      - 	Start   	 app|
+		'''.withPackage
+	
+		// when
+		val proposals = tclSnippet.proposals
+
+		// then
+		proposals => [
+			expect('application "path"').prefix.assertEquals('app')
+			reject('Stop application')
+		]
+	}
+
+	@Test
+	def void testElementProposal() {
+		// given
+		var tclSnippet = '''
+			# Some
+			
+			* some step
+			Component: GreetingApplication
+			- Read value from <|>
+		'''.withPackage
+
+		// when
+		val proposals = tclSnippet.proposals
+
+		// then
+		proposals => [
+			expect('Input').prefix.assertEquals('<')
+			expect('bar').prefix.assertEquals('<')
+			reject('Ok') // buttons have no 'Read value from' - fixture
+		]
+	}
+
+	@Test
+	def void testElementProposalNoClosingBracket() {
+		// given
+		var tclSnippet = '''
+			# Some
+			
+			* some step
+			Component: GreetingApplication
+			- Read value from <|
+		'''.withPackage
+
+		// when
+		val proposals = tclSnippet.proposals
+
+		// then
+		proposals => [
+			expect('Input>').prefix.assertEquals('<')
+			expect('bar>').prefix.assertEquals('<')
+			reject('Ok>') // buttons have no 'Read value from' - fixture
+		]
+	}
+
+	@Test
+	def void testIncompleteElementProposal() {
+		// given
+		var tclSnippet = '''
+			# Some
+			
+			* some step
+			Component: GreetingApplication
+			- Read value from <b|>
+		'''.withPackage
+
+		// when
+		val proposals = tclSnippet.proposals
+
+		// then
+		proposals => [
+			reject('Input')
+			expect('bar').prefix.assertEquals('b')
+			reject('Ok') 
+		]
+	}
+
+	@Test
+	def void testTrailingElementProposal() {
+		// given
+		var tclSnippet = '''
+			# Some
+			
+			* some step
+			Component: GreetingApplication
+			- Read value from <Input> |
+		'''.withPackage
+
+		// when
+		val proposals = tclSnippet.proposals
+
+		// then
+		proposals => [
```

Comment:
Are there any valid proposals at this point at all, i.e. should the assertion be that the list of proposals is empty?